cupy/indexing/generate.py

```diff
@@ -252,7 +252,57 @@ def ix_(*args):
 # TODO(okuta): Implement ravel_multi_index
 
 
-# TODO(okuta): Implement unravel_index
+def unravel_index(indices, dims, order='C'):
+    """Converts a flat index or array of flat indices into a tuple of coordinate arrays.
+
+    Args:
+        indices: An integer array whose elements are indices
+            into the flattened version of an array of dimensions :obj:`dims`.
+        dims: The shape of the array to use for unraveling indices.
+        order: Determines whether the indices should be viewed as indexing
+            in row-major (C-style) or column-major (Fortran-style) order.
+
+    Returns: tuple of ndarrays:
+        Each array in the tuple has the same shape as the indices array.
+
+    Examples
+    --------
+    >>> cupy.unravel_index(cupy.array([22, 41, 37]), (7, 6))
+    (array([3, 6, 6]), array([4, 5, 1]))
+    >>> cupy.unravel_index(cupy.array([31, 41, 13]), (7, 6), order='F')
+    (array([3, 6, 6]), array([4, 5, 1]))
+
+    .. seealso:: :func:`numpy.unravel_index`
+
+    """
+    if order == 'C':
+        dims = reversed(dims)
+    elif order == 'F':
+        pass
+    else:
+        raise TypeError("order not understood")
+
+    if not cupy.can_cast(indices, cupy.int64, "same_kind"):
+        raise TypeError(
+            "Iterator operand 0 dtype could not be cast "
+            "from dtype('{}') to dtype('{}') "
+            "according to the rule 'same_kind'".format(
+                indices.dtype, cupy.int64().dtype))
+
+    if (indices < 0).any():
+        raise ValueError("invalid entry in index array")
+
+    unraveled_coords = list()
```

Comment:
Use `[]`.

Reply:
OK. Is this coding rule documented in Chainer/CuPy?